Use latest sbt 1.0.0-RC2

dwijnand opened the pull request:
Fixes #372

dwijnand committed:
Fix #372: Use sbt 1.0.0-RC2

Fixes #372

dwijnand commented:
Does anyone understand why Drone is failing?

jvican commented:
What happens is that the sbt-protoc plugin for 1.0.0-RC2 is failing because of this error:
```

69s
607
protoc-jar: executing: [/tmp/protocjar3432661359923769209/bin/protoc.exe, --plugin=protoc-gen-scala=/tmp/protocbridge5776794677794621875, --scala_out=grpc:/drone/src/github.com/sbt/zinc/internal/zinc-persist/target/scala-2.12/src_managed/main, -I/drone/src/github.com/sbt/zinc/internal/zinc-persist/src/main/protobuf, -I/drone/src/github.com/sbt/zinc/internal/zinc-persist/target/protobuf_external, /drone/src/github.com/sbt/zinc/internal/zinc-persist/src/main/protobuf/schema.proto]
69s
608
java.lang.NoSuchMethodError: com.trueaccord.scalapb.Scalapb.registerAllExtensions(Lscalapb/com/google/protobuf/ExtensionRegistry;)V
69s
609
	at scalapb.ScalaPbCodeGenerator$.registerExtensions(ScalaPbCodeGenerator.scala:12)
69s
610
	at protocbridge.frontend.PluginFrontend$.runWithBytes(PluginFrontend.scala:52)
69s
611
	at protocbridge.frontend.PluginFrontend$.runWithInputStream(PluginFrontend.scala:67)
69s
612
	at protocbridge.frontend.PosixPluginFrontend$.$anonfun$prepare$1(PosixPluginFrontend.scala:25)
69s
613
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
69s
614
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653)
69s
615
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
69s
616
	at scala.util.Success.map(Try.scala:209)
69s
617
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
69s
618
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
69s
619
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
69s
620
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
69s
621
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
69s
622
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
69s
623
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
69s
624
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
69s
625
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
  
which makes the process run in the background forever. Drone shuts down the process when the build doesn't output something in 15 minutes.

dwijnand commented:
Ok, thanks. Do you know what's going on with scalapb and sbt-protoc? I just copied what you had done, but bumped to the latest release.

jvican commented on project/plugins.sbt:
```diff
@@ -1,14 +1,13 @@
-resolvers += Resolver.bintrayIvyRepo("scalacenter", "sbt-releases")
-addSbtPlugin(
-  ("com.thesamet" % "sbt-protoc" % "0.99.12")
-    .exclude("com.trueaccord.scalapb", "protoc-bridge_2.12"))
```
I remember having the problem you're experiencing. I fixed it by using here `protoc-bridge_2.12` instead of `protoc-bridge_2.10`. I don't know what may have triggered this change again... Try removing the exclusion line.

jvican commented on build.sbt:
```diff
@@ -74,17 +74,9 @@ def altPublishSettings: Seq[Setting[_]] =
-        )
-      streams.value.log.info("Publishing " + module + " to local repo: " + altLocalRepoName)
+      val module = new ivy.Module(moduleSettings)
+      val newConfig = config withResolverName altLocalRepoName withOverwrite false
```
Nitpick: I know you're a fan of infix, but can we have parenthesis for the first apply here? I find it difficult to read and parse in my mind.

dwijnand replied:
You mean

```scala
config.withResolverName(altLocalRepoName) withOverwrite false
```

?

jvican replied:
No, as you like infix, I think `(config withResolverName altLocalRepoName) withOverwrite false` would be better.

eed3si9n replied:
I am against the indiscriminate use of infix. Infixing should be limited to operator-like methods (e.g. collection library methods that accepts lambdas).

`config withResolverName altLocalRepoName withOverwrite false` is not readable. In my opinion, the only way this should be written is:

```scala
config
  .withResolverName(altLocalRepoName)
  .withOverwrite(false)
```

jvican commented:
sbt-protoc may be failing because of an error when it was published. @dwijnand If I were you, I would try to publish sbt-protoc myself to sbt's repo, making sure that:
1. Scalapb's version and dependencies correct. Mind the shaded version, may have something to do with this error.
2. For the selected scalapb version, there exists the interface that sbt-protoc is looking for via reflection.

jvican commented:
It would be great if we now add sbt-mima in this PR.

dwijnand commented:
If you could fix whatever this protobuf situation is, that would be lovely.

jvican committed:
Use Scala 2.12.3

We have to look into the reason why the shaded version of the compiler
plugin reports the following error and halts the build:

```
protoc-jar: executing: [/tmp/protocjar2818613521400194680/bin/protoc.exe, --plugin=protoc-gen-scala=/tmp/protocbridge4921434211196915044, --scala_out=grpc:/data/rw/code/scala/zinc/internal/zinc-persist/target/scala-2.12/src_managed/main, -I/data/rw/code/scala/zinc/internal/zinc-persist/src/main/protobuf, -I/data/rw/code/scala/zinc/internal/zinc-persist/target/protobuf_external, /data/rw/code/scala/zinc/internal/zinc-persist/src/main/protobuf/schema.proto]
java.lang.AbstractMethodError: Method scalapb/ScalaPbCodeGenerator$.run(Lcom/google/protobuf/compiler/PluginProtos$CodeGeneratorRequest;)Lcom/google/protobuf/compiler/PluginProtos$CodeGeneratorResponse; is abstract
	at scalapb.ScalaPbCodeGenerator$.run(ScalaPbCodeGenerator.scala)
	at protocbridge.frontend.PluginFrontend$.$anonfun$runWithBytes$1(PluginFrontend.scala:56)
	at scala.util.Try$.apply(Try.scala:209)
	at protocbridge.frontend.PluginFrontend$.runWithBytes(PluginFrontend.scala:54)
	at protocbridge.frontend.PluginFrontend$.runWithInputStream(PluginFrontend.scala:67)
	at protocbridge.frontend.PosixPluginFrontend$.$anonfun$prepare$1(PosixPluginFrontend.scala:25)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

jvican committed:
Upgrade scalapb version

jvican committed:
Exclude bridge and add shaded version

jvican commented:
Superseeded by #378. Thanks for getting the ball rolling @dwijnand. I'll continue your work.
